### PR TITLE
New Resource: `azurerm_api_management_identity_provider_google`

### PR DESCRIPTION
**Pull request by aqche**
Partially addresses: #5044

Adds the `azurerm_api_management_identity_provider_google` resource.

```
--- PASS: TestAccAzureRMApiManagementIdentityProviderGoogle_basic (1801.15s)
--- PASS: TestAccAzureRMApiManagementIdentityProviderGoogle_update (1862.55s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/apimanagement/tests 1862.64
```

### CODE REVIEW

File: azurerm/internal/services/apimanagement/resource_arm_api_management_identity_provider_google.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+			"api_management_name": azure.SchemaApiManagementName(),
+
+			"client_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"client_secret": {
```
Comment (katbyte) [inline azurerm/internal/services/apimanagement/resource_arm_api_management_identity_provider_google.go]: Could we change this to match what google calls this: 
```suggestion
			"api_ey": {
```

File: azurerm/internal/services/apimanagement/resource_arm_api_management_identity_provider_google.go [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+		Schema: map[string]*schema.Schema{
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"api_management_name": azure.SchemaApiManagementName(),
+
+			"client_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.NoEmptyStrings,
```
Comment (katbyte) [inline azurerm/internal/services/apimanagement/resource_arm_api_management_identity_provider_google.go]: Is there a validation we could do on this value?

File: azurerm/internal/services/apimanagement/resource_arm_api_management_identity_provider_google.go (hunk 1 above)
Comment (aqche) [reply azurerm/internal/services/apimanagement/resource_arm_api_management_identity_provider_google.go]: I believe both Azure and Google actually refer to the value as client secret. I'll update the documentation to reflect this.

**Comment by aqche** [comment]
@katbyte thanks for the review! made updates to the PR based on your comments. let me know what you think

File: azurerm/internal/services/apimanagement/resource_arm_api_management_identity_provider_google.go (hunk 1 above)
Comment (katbyte) [reply azurerm/internal/services/apimanagement/resource_arm_api_management_identity_provider_google.go]: If you look at the SDK docs (and presumably the API) it states:
```
// ClientSecret - Client secret of the Application in external Identity Provider, used to authenticate login request. For example, it is App Secret for Facebook login, API Key for Google login, Public Key for Microsoft.
```

Is this inaccurate and client secret is what you would look for on the google side?

Comment (aqche) [reply azurerm/internal/services/apimanagement/resource_arm_api_management_identity_provider_google.go]: Yep, it is called client secret as well on the Google side.

google

Comment (katbyte) [reply azurerm/internal/services/apimanagement/resource_arm_api_management_identity_provider_google.go]: perfect, thanks for that!

**Pull request merged**